corehq/apps/app_manager/models.py

```diff
@@ -986,9 +986,91 @@ def validate(self, value, required=True):
 
 
 class NavMenuItemMediaMixin(DocumentSchema):
+    """
+        Language-specific icon and audio.
+        Properties are map of lang-code to filepath
+
+        Convention (for old docs): Unless a language key exists in the map,
+        by default, media for it will be same as that pointed by 'default' key.
+        This is enforced by set/get methods contained in this class.
+
+    """
+    media_image = SchemaDictProperty(JRResourceProperty)
+    media_audio = SchemaDictProperty(JRResourceProperty)
+
+    @classmethod
+    def wrap(cls, data):
+        # ToDo - Remove after migration
+        for media_attr in ('media_image', 'media_audio'):
+            old_media = data.get(media_attr, None)
+            if old_media and isinstance(old_media, basestring):
+                new_media = {'default': old_media}
+                data[media_attr] = new_media
+
+        return super(NavMenuItemMediaMixin, cls).wrap(data)
+
+    @staticmethod
+    def _check_media_attribute(media_attr):
+        if media_attr not in ('media_image', 'media_audio'):
+            raise Exception("Unknown media attribute %s." % media_attr)
+
+    def _get_media_by_language(self, media_attr, lang):
+        self._check_media_attribute(media_attr)
+
+        media_dict = getattr(self, media_attr)
+        if not media_dict:
+            return None
+        default_media = media_dict.get('default', '')
+
+        return media_dict[lang] if lang in media_dict else default_media
+
+    @property
+    def default_media_image(self):
+        return self.icon_by_language('default')
+
+    @property
+    def default_media_audio(self):
+        return self.audio_by_language('default')
+
+    def icon_by_language(self, lang):
+        return self._get_media_by_language('media_image', lang)
+
+    def audio_by_language(self, lang):
+        return self._get_media_by_language('media_audio', lang)
+
+    def _set_media(self, media_attr, lang, media_path, default_lang=None):
+        """
+            Caller's responsibility to save doc.
+            Currently only called from the view which saves after all Edits
+        """
+        self._check_media_attribute(media_attr)
+
+        media_dict = getattr(self, media_attr) or {}
+        media_dict[lang] = media_path or ''
+        setattr(self, media_attr, media_dict)
```

Comment:
What the purpose of `media_path or ''`. When will `media_path` be `None`?

Also you can just go `self[media_attr][lang] = media_path or ''`